log/internal_logger.go

```diff
@@ -0,0 +1,56 @@
+package log
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+var (
+	analyticsServerURL = "https://bitrise-step-analytics.herokuapp.com"
+	netClient = http.Client{
+		Timeout: time.Second * 5,
+	}
+)
+
+// Message represents a line in a log
+type Message struct{
+	LogLevel string `json:"log_level"`
+	Message string `json:"message"`
+	Data map[string]interface{} `json:"data"`
+}
+
+// SetAnalyticsServerURL updates the the analytics server collecting the
+// logs. Default value is 'https://bitrise-step-analytics.herokuapp.com'
+func SetAnalyticsServerURL(url string) {
+	analyticsServerURL = url
+}
+
+// Internal sends the log message to the configured analytics server
+func (lm Message) Internal(stepID, tag string, data map[string]interface{}) {
+	lm.Data = make(map[string]interface{})
+	for k, v := range data {
+		lm.Data[k] = v
+	}
+
+	if v, ok := lm.Data["step_id"]; ok {
+		fmt.Printf("internal logger: data.step_id (%s) will be overriden with (%s) ", v, stepID)
+	}
+	if v, ok := lm.Data["tag"]; ok {
+		fmt.Printf("internal logger: data.tag (%s) will be overriden with (%s) ", v, tag)
+	}
+
+	lm.Data["step_id"] = stepID
+	lm.Data["tag"] = tag
+
+	var b bytes.Buffer
+	if err := json.NewEncoder(&b).Encode(lm); err != nil {
+		return
+	}
+
+	_, err := netClient.Post(analyticsServerURL + "/logs", "application/json", &b)
```

Comment:
NOTE: did not you like my idea about sending the log asynchronous?
anyway pls move this inside the if block

Reply:
Ouch -- I forgot about it. Fixing right away.

Reply:
In case this will be async, need to take care about finishing async tasks when the process finishes. So most likely sending an internal log in the last couple of lines of main, it won't be sent as the process finishes earlier than the async task.

Reply:
As discussed on Friday, we are not interested in handling this case now, as it is not a critical issue.